plasmapy/formulary/parameters.py

```diff
@@ -209,10 +209,14 @@ def Alfven_speed(B, density, ion="p+", z_mean=None):
 @utils.check_relativistic
 @utils.check_quantity(
     T_i={'units': u.K, 'can_be_negative': False},
-    T_e={'units': u.K, 'can_be_negative': False}
+    T_e={'units': u.K, 'can_be_negative': False},
+    n_e={'units': u.m ** -3, 'can_be_negative': False},
+    k={'units': u.m ** -1, 'can_be_negative': False}
     )
 def ion_sound_speed(T_e,
                     T_i,
+                    n_e=0*u.m**-3,
+                    k=0*u.m**-1,
```

Comment:
I'd use `None` as default values...something like

``` python
n_e = None,
k = None,
```

I think this would solve all your checking of default value issues.  At this point, if the user passes in a bad value (such as `n_e = 0.0 * u.m**-3`), then errors will be thrown accordingly, as they should.